web/styles/zulip.css

```diff
@@ -2218,16 +2221,23 @@ body:not(.spectator-view) {
             top: 0.05em;
         }
 
-        .dropdown-list-delete,
-        .dropdown-list-edit {
-            position: absolute;
-            top: 0;
-            right: 5px;
+        .dropdown-list-buttons {
+            display: flex;
             visibility: hidden;
         }
 
-        .dropdown-list-edit {
-            right: 30px;
+        .dropdown-list-item-name {
+            display: grid;
+            grid-template-columns: 1fr auto;
+            align-items: center;
+        }
+
+        .dropdown-list-item-description {
+            grid-column: 1;
+        }
+
+        &:has(.channel-privacy-type-icon) {
+            grid-template-columns: auto 1fr auto;
```

Comment:
This could be avoided with something like `minmax(0, auto)` on the main grid definition, I believe, so that the empty column just collapses to zero. That, too, depends on declaring named grid areas and placing items on them.